### PR TITLE
Document SSH requirements

### LINKED ISSUE #460: Document SSH expectations of kubeone
We should have `docs/ssh.md` that describe how kubeone is expecting to work with SSH, describing processed about ssh-agent and providing private key directly.

Evidences of confusion: #457 #458 and also in kubermatic slack this issue was brought up.

### PR DESCRIPTION
**Pull request by kron4eg**
Fixes #460 

```release-note
docs about ssh
```

### CODE REVIEW

File: docs/ssh.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+in order to cache it in ssh-agent memory for later use.
+
+KubeOne is able to contact ssh-agent via socket (environment variable
+`SSH_AUTH_SOCK`) and ask for authentication without getting unencrypted private
+key.
+
+## Providing SSH private keys directly, without ssh-agent
+
```
Comment (themue) [inline docs/ssh.md]: s/\n//

File: docs/ssh.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# KubeOne and SSH
+As kubeOne requires ssh access to control plane nodes, somehow ssh
```
Comment (xmudrii) [inline docs/ssh.md]: s/kubeOne/KubeOne

File: docs/ssh.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# KubeOne and SSH
```
Comment (xmudrii) [inline docs/ssh.md]: I'd probably name this something along `KubeOne SSH requirements` or `How KubeOne uses SSH`.

File: docs/ssh.md (hunk 2 above)
Comment (kron4eg) [reply docs/ssh.md]: 👍

File: docs/ssh.md (hunk 3 above)
Comment (kron4eg) [reply docs/ssh.md]: 👍

**Pull request merged**